src/lib/components/DeckGLMap/layers/wells/wellsLayer.ts

```diff
@@ -83,8 +171,62 @@ export default class WellsLayer extends CompositeLayer<
             })
         );
 
-        if (this.props.outline) return [outline, colors, highlight];
-        else return [colors, highlight];
+        const log_layer = new PathLayer<LogCurveDataType>(
+            this.getSubLayerProps({
+                id: "log_curve",
+                data: this.props.logData,
+                pickable: true,
+                widthScale: 10,
+                widthMinPixels: 1,
+                miterLimit: 100,
+                getPath: (d: LogCurveDataType): number[] => d.data[0],
+                getColor: (d: LogCurveDataType): RGBAColor[] =>
+                    getLogColor(d, this.props.logName),
+                getWidth: (d: LogCurveDataType): number | number[] | null =>
+                    this.props.logRadius || getLogWidth(d, this.props.logName),
+                updateTriggers: {
+                    getColor: [this.props.logName],
+                    getWidth: [this.props.logName, this.props.logRadius],
+                },
+            })
+        );
+
+        const layers: (GeoJsonLayer<Feature> | PathLayer<LogCurveDataType>)[] =
+            [colors, highlight];
+        if (this.props.outline) {
+            layers.splice(0, 0, outline);
+        }
+        if (this.props.logCurves) {
+            layers.splice(1, 0, log_layer);
+        }
+        return layers;
+    }
+
+    getPickingInfo({ info }) {
+        if (info.object == null || info.object.data == undefined) return info;
+
+        const trajectory = info.object.data[0];
+
+        let min_d = 99999;
+        let vertex_index = 0;
+        for (let i = 0; i < trajectory.length; i++) {
+            const d = squared_distance(trajectory[i], info.coordinate);
+            if (d > min_d) continue;
+
+            vertex_index = i;
+            min_d = d;
+        }
```

Comment:
This function will be called for every mouse move, so I'm a bit concerned that this approach will be too slow. We can keep it for now and profile later.